src/main/java/io/strimzi/kafka/access/KafkaAccessReconciler.java

```diff
@@ -4,35 +4,130 @@
         this.kubernetesClient = kubernetesClient;
+        final Base64.Encoder encoder = Base64.getEncoder();
+        commonSecretData.put(TYPE_SECRET_KEY, encoder.encodeToString(TYPE_SECRET_VALUE.getBytes(StandardCharsets.UTF_8)));
+        commonSecretData.put(PROVIDER_SECRET_KEY, encoder.encodeToString(PROVIDER_SECRET_VALUE.getBytes(StandardCharsets.UTF_8)));
+        commonSecretLabels.put(Utils.MANAGED_BY_LABEL_KEY, Utils.KAFKA_ACCESS_LABEL_VALUE);
     }
 
     @Override
-    public UpdateControl<KafkaAccess> reconcile(KafkaAccess kafkaAccess, Context context) {
-        return UpdateControl.noUpdate();
+    public UpdateControl<KafkaAccess> reconcile(final KafkaAccess kafkaAccess, final Context context) {
+        final String kafkaAccessName = kafkaAccess.getMetadata().getName();
+        final String kafkaAccessNamespace = kafkaAccess.getMetadata().getNamespace();
+        LOGGER.info(String.format("Reconciling KafkaAccess %s/%s", kafkaAccessNamespace, kafkaAccessName));
+        final Map<String, String> data  = new HashMap<>(commonSecretData);
+        context.getSecondaryResource(Secret.class)
+            .ifPresentOrElse(secret -> {
+                final Map<String, String> currentData = secret.getData();
+                if (!data.equals(currentData)) {
+                    kubernetesClient.secrets()
+                            .inNamespace(kafkaAccessNamespace)
+                            .withName(kafkaAccessName)
+                            .edit(s -> new SecretBuilder(s).withData(data).build());
+                }
+            }, () -> kubernetesClient
+                    .secrets()
+                    .inNamespace(kafkaAccessNamespace)
+                    .create(
+                            new SecretBuilder()
+                                    .withType(SECRET_TYPE)
+                                    .withNewMetadata()
+                                    .withName(kafkaAccessName)
+                                    .withLabels(commonSecretLabels)
+                                    .withOwnerReferences(
+                                            new OwnerReferenceBuilder()
+                                                    .withApiVersion(kafkaAccess.getApiVersion())
+                                                    .withKind(kafkaAccess.getKind())
+                                                    .withName(kafkaAccessName)
+                                                    .withUid(kafkaAccess.getMetadata().getUid())
+                                                    .withBlockOwnerDeletion(false)
+                                                    .withController(false)
+                                                    .build()
+                                    )
+                                    .endMetadata()
+                                    .withData(data)
+                                    .build()
+                    )
+            );
```

Comment:
couldn't we factor out a kind of `createOrUpdate` method for doing this part, making "lighter" the reconcile method?

Reply:
I've still got quite a few updates to make to the reconcile method to include parsing the Kafka and KafkaUser so I think I would prefer to wait until the PR where I add those. Otherwise any refactor now might not make sense once those changes are in. I will keep it in mind for the next PR though.